### PR TITLE
Resolved Support Version 2.0.0 changes for V1/regard-application

### LINKED ISSUE #99: Include domain-name support for the service regard-application
**Note : This issue can be taken after completing the issue https://github.com/openBackhaul/ApplicationPattern/issues/350**
In the initial CONFIGfile , there will not be any attributes related to domain-name.
In the server-side logic , the code needs to be enhanced so that if a domain-name attribute can be added via a service , then the server-side code should include the domain-name attribute along with the values and delete the ip-address attribute at the same time and vice versa.

### LINKED ISSUE #100: Include protocol support for the service regard-application
**Note : This issue can be taken after completing the issue https://github.com/openBackhaul/ApplicationPattern/issues/381**
As a part of the request body , there is a new attribute “protocol” added. 
Include support to read and store the protocol attribute in the tcp-client instance.
Note : Incase of an existing instance update the incoming values if not create a new instance.

### LINKED ISSUE #101: Include uuid syntax for the redirect-service-request-information attribute for the service regard-application
As a part of regard-application , for the basic service “redirect-service-request-information” a new operation-client will be created for the regarded application.
As per the specification , the reserved uuid syntax for this operationName is 
```
<own application abbreviation>-<release-number-of-own-separated-by-hypens>-op-c-im-<application abbreviation>-<release-number-separated-by-hypens>-004
```
Create a OperationClient if not available. Update the OperationClient configuration if the client is already available.

### LINKED ISSUE #102: Support Version 2.0.0 changes for V1/regard-application
1. Generate server stub for the version 2.0.0 specification 
2. If the api/openapi.yaml is not updated as per the published version 2.0.0 , then copy the api/openapi.yaml file from the generated stub to the source code
3. no change required in the method signature

Following tasks needs to be completed to support /v1/bequeath-your-data-and-die version 2.0.0 , 

- [ ] https://github.com/openBackhaul/OamLog/issues/99
- [ ] https://github.com/openBackhaul/OamLog/issues/100
- [ ] https://github.com/openBackhaul/OamLog/issues/101

### PR DESCRIPTION
**Pull request by ManasaBM1**
Fixes #102
Fixes #99
Fixes #100
Fixes #101

### CODE REVIEW

**Comment by ManasaBM1** [comment]
Reviewer is @DanaSunal

**Review by DanaSunal — CHANGES_REQUESTED** [review]
Please cleanup the commit. StartAppInGenericRepresentation should be already merged, some code should be within PR #172 .

**Comment by ManasaBM1** [comment]
done

File: server/service/IndividualServicesService.js [hunk 1]
```diff
@@ -356,27 +357,24 @@ exports.regardApplication = function (body, user, originator, xCorrelator, trace
...
-      let operationList = [
-        oamRequestOperation
-      ];
+    
       let logicalTerminatinPointConfigurationInput = new LogicalTerminatinPointConfigurationInput(
         applicationName,
         releaseNumber,
         applicationAddress,
         applicationPort,
         operationList
```
Comment (DanaSunal) [inline server/service/IndividualServicesService.js:375]: operationList does not exist

File: server/service/IndividualServicesService.js [hunk 2]
```diff
@@ -36,6 +36,7 @@ const serviceRecordProfile = require('onf-core-model-ap/applicationPattern/onfMo
 const ProfileCollection = require('onf-core-model-ap/applicationPattern/onfModel/models/ProfileCollection');
 const Profile = require('onf-core-model-ap/applicationPattern/onfModel/models/Profile');
 const OamRecordProfile = require('onf-core-model-ap/applicationPattern/onfModel/models/profile/OamRecordProfile');
+const individualServicesOperationsMapping = require('./individualServices/IndividualServicesOperationsMapping');
```
Comment (DanaSunal) [inline server/service/IndividualServicesService.js:39]: Move this to PR #172

File: server/service/IndividualServicesService.js [hunk 3]
```diff
@@ -356,27 +357,24 @@ exports.regardApplication = function (body, user, originator, xCorrelator, trace
...
+    
       let logicalTerminatinPointConfigurationInput = new LogicalTerminatinPointConfigurationInput(
         applicationName,
         releaseNumber,
         applicationAddress,
         applicationPort,
         operationList
       );
-      let logicalTerminationPointconfigurationStatus = await LogicalTerminationPointService.findOrCreateApplicationInformationAsync(
+      let logicalTerminationPointconfigurationStatus = await LogicalTerminationPointService.createOrUpdateApplicationInformationAsync(
```
Comment (DanaSunal) [inline server/service/IndividualServicesService.js:377]: why was this change done?

File: server/service/individualServices/individualServicesOperationsMapping.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+module.exports.individualServicesOperationsMapping =
+{
+    "/v1/regard-application":
+    {
+        "redirect-oam-request-information": {
+            "api-segment": "im",
+            "sequence": "004"
+        }
+    }
+}
```
Comment (DanaSunal) [inline server/service/individualServices/individualServicesOperationsMapping.js:10]: Move this to PR #172

**Review by DanaSunal — CHANGES_REQUESTED** [review]

File: server/service/IndividualServicesService.js (hunk 1 above)
Comment (ManasaBM1) [reply server/service/IndividualServicesService.js]: removed

File: server/service/IndividualServicesService.js (hunk 3 above)
Comment (ManasaBM1) [reply server/service/IndividualServicesService.js]: removed ..revered to new changes in recent PR

File: server/service/individualServices/individualServicesOperationsMapping.js (hunk 4 above)
Comment (ManasaBM1) [reply server/service/individualServices/individualServicesOperationsMapping.js:10]: done

File: server/service/IndividualServicesService.js (hunk 2 above)
Comment (ManasaBM1) [reply server/service/IndividualServicesService.js:39]: done

**Review by DanaSunal — APPROVED** [review]

**Pull request merged**